File: C++/mathFunctions.cpp
#include <iostream>

using namespace std;

// Function to add two numbers
int add(int a, int b) {
    return a + b;
}

// Function to subtract two numbers
int subtract(int a, int b) {
    return a - b;
}

// Function to multiply two numbers
int multiply(int a, int b) {
    return a * b;
}

// Function to divide two numbers
int divide(int a, int b) {
    return a / b;
}

// Function to find remainder of two numbers
int remainder(int a, int b) {
    return a % b;
}

// Function to find power of a number
int power(int a, int b) {
    int result = 1;
    for (int i = 0; i < b; i++) {
        result *= a;
    }
    return result;
}

// Function to find square root of a number
int squareRoot(int a) {
    int result = 1;
    while (result * result <= a) {
        result++;
    }
    return result - 1;
}

// Function to absolute value of a number
int absolute(int a) {
    if (a < 0) {
        return -a;
    }
    return a;
}

// Function to calculate hypotenuse of a right angled triangle
int hypotenuse(int a, int b) {
    return squareRoot(a * a + b * b);
}

// Main function
int main() {
    // decalre two numbers in the same line
    int a, b;

    // capture first number
    cout << "Enter first number: ";
    cin >> a;

    // capture second number
    cout << "Enter second number: ";
    cin >> b;

    // Add two numbers
    cout << "Sum of " << a << " and " << b << " is " << add(a, b) << endl; // a+b = a - (-b)

    // Subtract two numbers
    cout << "Difference of " << a << " and " << b << " is " << subtract(a, b) << endl; // a-b = a + (-b)

    // Multiply two numbers
    cout << "Product of " << a << " and " << b << " is " << multiply(a, b) << endl; // a*b = a + a + a + ... + a (b times)

    // Divide two numbers
    cout << "Quotient of " << a << " and " << b << " is " << divide(a, b) << endl; // a/b = a - (a%b)

    // Find remainder of two numbers
    cout << "Remainder of " << a << " and " << b << " is " << remainder(a, b) << endl; // a%b = a - (a/b)*b

    // Find power of a number
    cout << "Power of " << a << " to " << b << " is " << power(a, b) << endl; // a^b = a*a*a*...*a (b times)

    // Find square root of a number
    cout << "Square root of " << a << " is " << squareRoot(a) << endl; // sqrt(a) = a - (a%1)*1

    // Find absolute value of a number
    cout << "Absolute value of " << a << " is " << absolute(a) << endl; // |a| = a - (-a)

    // Find hypotenuse of a right angled triangle
    cout << "Hypotenuse of a right angled triangle with sides " << a << " and " << b << " is " << hypotenuse(a, b) << endl; // sqrt(a^2 + b^2)

    return 0;
}

File: TS/mathFunctions.ts
// function to add two variables

function add(a: number, b: number): number {
  return a + b;
}

console.log(`add(2, 3) = ${add(2, 3)}`); // add(2, 3) = 5

// function to subtract two variables

function subtract(a: number, b: number): number {
  return a - b;
}

console.log(`subtract(2, 3) = ${subtract(2, 3)}`); // subtract(2, 3) = -1

// function to multiply two variables

function multiply(a: number, b: number): number {
  return a * b;
}

console.log(`multiply(2, 3) = ${multiply(2, 3)}`); // multiply(2, 3) = 6

// function to divide two variables

function divide(a: number, b: number): number {
  return a / b;
}

console.log(`divide(2, 3) = ${divide(2, 3)}`); // divide(2, 3) = 0.6666666666666666
console.log(`divide(2, 3) = ${divide(2, 3).toFixed(2)}`); // divide(2, 3) = 0.67
console.log(`divide(1, 0) = ${divide(1, 0)}`); // divide(1, 0) = Infinity

// function to calculate the remainder of two variables

function remainder(a: number, b: number): number {
  return a % b;
}

console.log(`remainder(2, 3) = ${remainder(2, 3)}`); // remainder(2, 3) = 2
console.log(`remainder(2, 3) = ${remainder(2, 3).toFixed(2)}`); // remainder(2, 3) = 2.00
console.log(`remainder(1, 0) = ${remainder(1, 0)}`); // remainder(1, 0) = NaN

// function to calculate the power of two variables

function power(a: number, b: number): number {
  return a ** b;
}

console.log(`power(2, 3) = ${power(2, 3)}`); // power(2, 3) = 8

// function to calculate the square root of a variable

function squareRoot(a: number): number {
  return Math.sqrt(a);
}

console.log(`squareRoot(2) = ${squareRoot(2)}`); // squareRoot(2) = 1.4142135623730951
console.log(`squareRoot(2) = ${squareRoot(2).toFixed(2)}`); // squareRoot(2) = 1.41
console.log(`squareRoot(-2) = ${squareRoot(-2)}`); // squareRoot(-2) = NaN

// function to calculate the absolute value of a variable

function absoluteValue(a: number): number {
  return Math.abs(a);
}

console.log(`absoluteValue(2) = ${absoluteValue(2)}`); // absoluteValue(2) = 2

// function to calculate the hypotenuse of a right triangle

function hypotenuse(a: number, b: number): number {
  return Math.hypot(a, b);
}

console.log(`hypotenuse(2, 3) = ${hypotenuse(2, 3)}`); // hypotenuse(2, 3) = 3.605551275463989